urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/presentation/SearchListMovieAndCreateBooking_CLI.java

```diff
@@ -0,0 +1,474 @@
+				tryAgain = true;
+				System.out.println("Invalid Movie Code. Try again.");
+				System.out.println();
+			}		
+		} while(tryAgain);
+		
+		printMovieInfo(movie); // Print Movie Details and all the ShowTimes
+		
+		do {
+			tryAgain = false;
+			System.out.print("Please select showtime (the number between \'[ ]\') (-1 to go back) : ");
+			showTime_index = ConsoleReader.readIntInput();
+			
+			if(showTime_index == -1) {
+				return;
+			}
+			
+			if(showTime_index < 1 || showTime_index > _showTimes.size()) {
+				tryAgain = true;
+				System.out.println("Invalid number. Try again.");
+			}
+			else {			
+				showTime = _showTimes.get(showTime_index-1); //This showTimes is construted when we list the showTimes. Not from the BL.
+				
+				if(showTimeBL.isFullyBooked(showTime)) {
+					System.out.println();
+					System.out.println("===========================================================================");
+					System.out.println("| We\'re sorry, the selected session is sold out. Please pick another one! |");
+					System.out.println("===========================================================================");
+					System.out.println();
+					tryAgain = true;
+					continue;
+				}
+				
+				printSeats(showTime);
+			}
+		} while(tryAgain);
+		
+		boolean isValidSeatNos;
+		do {
+			System.out.print("Please enter seat no. (e.g. G1 G2 ...) : ");
+			seats = ConsoleReader.readSeatNumbers();
+			String[] _seats = seats.split(" ");
+			isValidSeatNos = seatBL.validateSeatNumbers(showTime, _seats);
+			
+			if(!isValidSeatNos) {
+				System.out.println("Invalid seat numbers. Please choose again.");
+			}
+			else if(_seats.length > 10) {
+				System.out.println("You cannot book more than 10 tickets per transaction. Please select seats again.");
+				isValidSeatNos = false;
+			}
+			System.out.println();
+		} while(!isValidSeatNos);
+		
+		//Ask for movieGoer info
+		String movieGoerName;
+		String movieGoerMobileNo;
+		String movieGoerEmail;
+		Date dateOfBirth = null;
+		
+		System.out.println();
+		System.out.println("Please Enter Your Details");
+		System.out.println("=========================");
+		
+		System.out.print("Name : ");
+		movieGoerName = ConsoleReader.readString();
+		
+		System.out.print("Date of Birth (DD/MM/YYYY) : ");
+		dateOfBirth = ConsoleReader.readDateInput();
+		
+		System.out.print("Email : ");
+		movieGoerEmail = ConsoleReader.readString();
+		
+		System.out.print("Mobile No : ");
+		movieGoerMobileNo = ConsoleReader.readString();
+		
+		movieGoer = movieGoerBL.createMovieGoer(movieGoerName, movieGoerEmail, movieGoerMobileNo, dateOfBirth);
+		
+		printBookingSummary(showTime, movieGoer, seats);
+	}
+	
+	public void printBookingSummary(ShowTime showTime, MovieGoer movieGoer, String seats) {
+		Booking booking;
+		Cinema cinema = showTime.getCinema();
+		Cineplex cineplex = cinema.getCineplex();
+		Movie movie = showTime.getMovie();
+		SimpleDateFormat dateFormat = new SimpleDateFormat("dd/MM/yyyy");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.